src/osd/OpRequest.h

```diff
@@ -46,6 +46,7 @@ struct OpRequest : public TrackedOp {
   bool need_skip_handle_cache() const { return op_info.need_skip_handle_cache(); }
   bool need_skip_promote() const { return op_info.need_skip_promote(); }
   bool allows_returnvec() const { return op_info.allows_returnvec(); }
+  bool only_stat_read() const { return op_info.only_stat_read(); }
```

Comment:
Instead of only_stat_read(), let's add may_read_data().

```
  /// Returns true iff op performs a read (including of the object_info)
  bool may_read() const { ... }

  /**
   * may_read_data()
   * 
   * Returns true iff op reads information other than the object_info.  Requires that the
   * osd flush any prior writes prior to servicing this op.  Includes any information not
   * cached by the osd in the object_info or snapset.
   */
  bool may_read_data() const { ... }
```

may_read() && !may_read_data() would be equivalent to only_stat_read().

Reply:
OK. I'll update commit soon.